src/inmanta/agent/handler.py

```diff
@@ -288,6 +299,10 @@ def action_id(self) -> uuid.UUID:
     def status(self) -> Optional[const.ResourceState]:
         return self._status
 
+    @property
+    def blocked(self) -> Optional[BlockedStatus]:
+        return self._blocked
+
```

Comment:
I think I see. Is it to be able to carry the differentiation over the wire to the scheduler? If so, I think it might be cleaner to simply add a new state to `const.ResourceState`. On the other hand, if we do that we have to carefully check where we expose values of that enum and whether we need to take action anywhere.

@wouterdb thoughts?

Reply:
We can't change const.ResourceState I think.

We are phasing it out, it leaks out of the api on all sides, I would not do that.

Reply:
How do we see this handler context evolving over time? It wouldn't have the `const.ResourceState` anymore, so how would it communicate that it wants to skip? I ask because I'm hesitant to add a field now, one that feels out of place, unless it will come to fit into the future plans.

Reply:
removed the new entry in `const.ResourceState` and created `ContextResourceState` to address this issue.